mrmustard/lab_dev/circuit_components.py

```diff
@@ -12,8 +12,95 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""
+r"""
 A base class for the components of quantum circuits.
+
+``CircuitComponent``\s bring together two types of information:
+
+* That contained in the ``Wires`` object, which identifies the modes spanned by a component.
+* That contained in the ``Representation``, which provides a mathematical description of a
+  component.
+Together, wires and representation uniquely define individual components, as well as their
+interaction with other components.
+
+This example shows how to define components with the same wires and representations as popular
+states and gates.
+
+.. code-block ::
+
+    >>> from mrmustard.lab_dev import *
+    >>> from mrmustard.physics.triples import vacuum_state_Abc, beamsplitter_gate_Abc, attenuator_Abc
+    >>> from mrmustard.physics.fock import fock_state
+    >>> from mrmustard.physics.representations import Bargmann, Fock
+    >>> import numpy as np
+
+    >>> # the vacuum state on mode `0`
+    >>> vac_rep = Bargmann(*vacuum_state_Abc(n_modes=1))
+    >>> vac_modes = (0,)
+    >>> vac = CircuitComponent("my_vac", vac_rep, modes_out_ket=vac_modes)
+
+    >>> # the number state on mode `3`
+    >>> num_rep = Fock(fock_state(n=1))
+    >>> num_modes = (3,)
+    >>> number = CircuitComponent("my_num", num_rep, modes_out_ket=num_modes)
+
+    >>> # the 50/50 beam splitter on modes `(8, 19)`
+    >>> bs_rep = Bargmann(*beamsplitter_gate_Abc(theta=np.pi/4))
+    >>> bs_modes = (8, 19)
+    >>> bs = CircuitComponent("my_bs", bs_rep, modes_out_ket=bs_modes, modes_in_ket=bs_modes)
+
+    >>> # the attenuator on mode `16`
+    >>> att_rep = Bargmann(*attenuator_Abc(0.9))
+    >>> att_modes = (16,)
+    >>> att = CircuitComponent("my_att", att_rep, att_modes, att_modes, att_modes, att_modes)
```

Comment:
To me it is a bit confusing why we repeat att_modes 4 times. I guess it is a channel and thus has two in and two out wires. For the other components it was more clear since there was a modes_out/in_ket statement.

Reply:
Nice catch!